lib/logflare/backends/adaptor/postgres_adaptor/supervisor.ex

```diff
@@ -0,0 +1,32 @@
+defmodule Logflare.Backends.Adaptor.PostgresAdaptor.Supervisor do
+  use Supervisor
+
+  alias Logflare.Backends.Backend
+
+  @repo_sup __MODULE__.Repos
+  @registry Logflare.Backends.SourceRegistry
+
+  def start_link(args) do
+    Supervisor.start_link(__MODULE__, args)
+  end
+
+  def init(_args) do
+    children = [
+      {DynamicSupervisor, strategy: :one_for_one, name: @repo_sup}
+    ]
+
+    Supervisor.init(children, strategy: :one_for_one)
+  end
+
+  def start_child(spec),
+    do: DynamicSupervisor.start_child(@repo_sup, spec)
+
+  def get(%Backend{id: id}) do
+    case Registry.lookup(@registry, {__MODULE__, id}) do
+      [{pid, meta}] -> {:ok, pid, meta}
+      [] -> :error
+    end
+  end
+
+  def via(%Backend{id: id}), do: {:via, Registry, {@registry, {__MODULE__, id}}}
```

Comment:
doc + spec

Reply:
would also prefer to use `Backends.via_source/3` in general to ensure that via ids are consistent while we're doing v1->v2 transition

Reply:
I will take a look.

Reply:
The thing is, where I use `Supervisor.via/1` I do not have access to `Source` as it is not relevant there (in `SharedRepo` start). This mean that I cannot use `via_source` as it expect me to pass `Source.t()`. This also would mean that multiple sources cannot reuse the same backend connection and instead each would spawn their own connection.